code/game/machinery/washing_machine.dm

```diff
@@ -14,306 +13,401 @@
-					J.item_color = wash_color
-					J.name = new_jumpsuit_name
-					J.desc = new_desc
-			if(new_glove_icon_state && new_glove_item_state && new_glove_name)
-				for(var/obj/item/clothing/gloves/color/G in src)
-					if(!G.dyeable)
-						continue
-					G.item_state = new_glove_item_state
-					G.icon_state = new_glove_icon_state
-					G.item_color = wash_color
-					G.name = new_glove_name
-					if(!istype(G, /obj/item/clothing/gloves/color/black/thief))
-						G.desc = new_desc
-			if(new_shoe_icon_state && new_shoe_name)
-				for(var/obj/item/clothing/shoes/S in src)
-					if(!S.dyeable)
-						continue
-					if(S.chained)
-						S.chained = FALSE
-						S.slowdown = SHOES_SLOWDOWN
-						new /obj/item/restraints/handcuffs(src)
-					S.icon_state = new_shoe_icon_state
-					S.item_color = wash_color
-					S.name = new_shoe_name
-					S.desc = new_desc
-			if(new_bandana_icon_state && new_bandana_name)
-				for(var/obj/item/clothing/mask/bandana/M in src)
-					if(!M.dyeable)
-						continue
-					M.item_state = new_bandana_item_state
-					M.icon_state = new_bandana_icon_state
-					M.item_color = wash_color
-					M.name = new_bandana_name
-					M.desc = new_desc
-			if(new_sheet_icon_state && new_sheet_name)
-				for(var/obj/item/bedsheet/B in src)
-					B.icon_state = new_sheet_icon_state
-					B.item_color = wash_color
-					B.item_state = new_sheet_item_state
-					B.name = new_sheet_name
-					B.desc = new_desc
-			if(new_softcap_icon_state && new_softcap_name)
-				for(var/obj/item/clothing/head/soft/H in src)
-					if(!H.dyeable)
-						continue
-					H.icon_state = new_softcap_icon_state
-					H.item_color = wash_color
-					H.name = new_softcap_name
-					H.desc = new_desc
-		QDEL_NULL(crayon)
-
-
-	if(locate(/mob) in src)
-		state = CLOSED_BLOODY
-		gibs_ready = TRUE
-	else
-		state = CLOSED_FULL
-	update_icon(UPDATE_ICON_STATE)
+/obj/machinery/washing_machine/attack_hand(mob/user)
+	if(washing && !door_open)
+		to_chat(user, "<span class='warning'>[src]'s door is shut, you will need to wait until the end of the cycle.</span>")
+		return
+	toggle_door()
 
-/obj/machinery/washing_machine/update_icon_state()
-	icon_state = "wm_[state][panel]"
 
 /obj/machinery/washing_machine/attackby(obj/item/W, mob/user, params)
+	if(user.a_intent == INTENT_HARM)
+		return ..()
 	if(default_unfasten_wrench(user, W))
 		return
-	if(istype(W, /obj/item/toy/crayon) || istype(W, /obj/item/stamp))
-		if(state in list(OPEN_EMPTY, OPEN_FULL))
-			if(!crayon)
-				user.drop_item()
-				crayon = W
-				crayon.forceMove(src)
-				update_icon(UPDATE_ICON_STATE)
-			else
-				return ..()
-		else
-			return ..()
-	else if(istype(W, /obj/item/grab))
-		if(state == OPEN_EMPTY)
-			var/obj/item/grab/G = W
-			if(ishuman(G.assailant) && iscorgi(G.affecting))
-				G.affecting.forceMove(src)
+	if(default_deconstruction_screwdriver(user, null, null, W))
+		return
+	if(default_deconstruction_crowbar(user, W, FALSE))
+		return
+	if(istype(W, /obj/item/soap))
```

Comment:
This shouldn't be a direct check against soap imo. Any cleaning_act should do. You may have to refactor how cleaning_act is called though in the attack chain for this though which might be out of scope.

Reply:
There's no way to accomplish what you're asking without a type check against soap. Either I perform a typecheck here and redirect the attack chain or the soap gets put into the washing machine through `attempt_insert()`